src/measure_theory/vector_measure.lean

```diff
@@ -618,6 +630,181 @@ begin
+    { exact ha₁.inter (measurable_set.disjointed hf₁ n) },
+    { exact pairwise_disjoint_on_inter disjoint_disjointed } },
+  { intro n, exact (ha₁.inter (measurable_set.disjointed hf₁ n)) },
+  { exact λ n, ha₁.inter (measurable_set.disjointed hf₁ n) }
+end
+
+lemma restrict_le_restrict_encodable_Union [encodable β] {f : β → set α}
+  (hf₁ : ∀ b, measurable_set (f b)) (hf₂ : ∀ b, v ≤[f b] w) :
+  v ≤[⋃ b, f b] w :=
+begin
+  rw ← encodable.Union_decode₂,
+  refine restrict_le_restrict_Union v w _ _,
+  { intro n, measurability },
+  { intro n,
+    cases encodable.decode₂ β n with b,
+    { simp },
+    { simp [hf₂ b] } }
+end
+
+lemma restrict_le_restrict_union
+  (hi₁ : measurable_set i) (hi₂ : v ≤[i] w)
+  (hj₁ : measurable_set j) (hj₂ : v ≤[j] w) :
+  v ≤[i ∪ j] w :=
+begin
+  rw union_eq_Union,
+  refine restrict_le_restrict_encodable_Union v w _ _,
+  { measurability },
+  { rintro (_ | _); simpa }
+end
+
+end
+
+section
+
+variables {M : Type*} [topological_space M] [ordered_add_comm_monoid M]
+variables (v w : vector_measure α M) {i j : set α}
+
+lemma zero_le_restrict_empty : 0 ≤[∅] v :=
+begin
+  intros j hj,
+  rw [restrict_empty, restrict_empty]
+end
+
+lemma restrict_empty_le_zero : v ≤[∅] 0 :=
+begin
+  intros j hj,
+  rw [restrict_empty, restrict_empty]
+end
+
+lemma nonneg_of_zero_le_restrict (hi₂ : 0 ≤[i] v) :
+  0 ≤ v i :=
+begin
+  by_cases hi₁ : measurable_set i,
+  { exact (restrict_le_restrict_iff _ _ hi₁).1 hi₂ hi₁ set.subset.rfl },
+  { rw v.not_measurable hi₁ },
+end
+
+lemma nonpos_of_restrict_le_zero (hi₂ : v ≤[i] 0) :
+  v i ≤ 0 :=
+begin
+  by_cases hi₁ : measurable_set i,
+  { exact (restrict_le_restrict_iff _ _ hi₁).1 hi₂ hi₁ set.subset.rfl },
+  { rw v.not_measurable hi₁ }
+end
+
+lemma zero_le_restrict_not_measurable (hi : ¬ measurable_set i) :
+  0 ≤[i] v :=
+begin
+  rw [restrict_zero, restrict_not_measurable _ hi],
+  exact le_refl _,
+end
+
+lemma restrict_le_zero_of_not_measurable (hi : ¬ measurable_set i) :
+  v ≤[i] 0 :=
+begin
+  rw [restrict_zero, restrict_not_measurable _ hi],
+  exact le_refl _,
+end
+
+lemma measurable_of_not_zero_le_restrict (hi : ¬ 0 ≤[i] v) : measurable_set i :=
+not.imp_symm (zero_le_restrict_not_measurable _) hi
+
+lemma measurable_of_not_restrict_le_zero (hi : ¬ v ≤[i] 0) : measurable_set i :=
+not.imp_symm (restrict_le_zero_of_not_measurable _) hi
+
+lemma zero_le_restrict_subset (hi₁ : measurable_set i) (hi₂ : 0 ≤[i] v) (hij : j ⊆ i) :
+  0 ≤[j] v :=
+restrict_le_restrict_of_subset_le _ _
+  (λ k hk₁ hk₂, (restrict_le_restrict_iff _ _ hi₁).1 hi₂ hk₁ (set.subset.trans hk₂ hij))
+
+lemma restrict_le_zero_subset (hi₁ : measurable_set i) (hi₂ : v ≤[i] 0) (hij : j ⊆ i) :
+  v ≤[j] 0 :=
+restrict_le_restrict_of_subset_le _ _
+  (λ k hk₁ hk₂, (restrict_le_restrict_iff _ _ hi₁).1 hi₂ hk₁ (set.subset.trans hk₂ hij))
+
+lemma not_zero_le_restrict_subset (hi : ¬ 0 ≤[i] v) (h : i ⊆ j) (hj : measurable_set j) :
+  ¬ 0 ≤[j] v :=
+λ h', hi $ (zero_le_restrict_subset _) hj h' h
```

Comment:
If you put the `≤[i]` argument last, I think you can prove these simply with `mt`.

Reply:
I don't quite understand what you mean here

Reply:
I mean if you change the argument order of the above lemmas to
```lean
lemma zero_le_restrict_subset (hi₁ : measurable_set i) (hij : j ⊆ i) (hi₂ : 0 ≤[i] v) :
  0 ≤[j] v :=
restrict_le_restrict_of_subset_le _ _
  (λ k hk₁ hk₂, (restrict_le_restrict_iff _ _ hi₁).1 hi₂ hk₁ (set.subset.trans hk₂ hij))

lemma restrict_le_zero_subset (hi₁ : measurable_set i) (hij : j ⊆ i) (hi₂ : v ≤[i] 0) :
  v ≤[j] 0 :=
restrict_le_restrict_of_subset_le _ _
  (λ k hk₁ hk₂, (restrict_le_restrict_iff _ _ hi₁).1 hi₂ hk₁ (set.subset.trans hk₂ hij))
```
Then these lemmas are trivial applications of `mt`, suggesting you may not need them at all:

```lean
lemma not_zero_le_restrict_subset (hj : measurable_set j) (h : i ⊆ j) : ¬ 0 ≤[i] v → ¬ 0 ≤[j] v :=
mt (zero_le_restrict_subset _ hj h)

lemma not_restrict_le_zero_subset (hj : measurable_set j) (h : i ⊆ j) : ¬ v ≤[i] 0 → ¬ v ≤[j] 0 :=
mt (restrict_le_zero_subset _ hj h)
```

Reply:
I've removed these lemmas now :)